pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -100,6 +102,9 @@ type CertificateManager interface {
 	LoadTrustedBundle(context.Context, client.Client, string) (certificatemanagement.TrustedBundleRO, error)
 	// LoadMultiTenantTrustedBundleWithRootCertificates loads an existing trusted bundle with system root certificates to pass to render.
 	LoadMultiTenantTrustedBundleWithRootCertificates(context.Context, client.Client, string) (certificatemanagement.TrustedBundleRO, error)
+	// SignCertificate signs a certificate using its own private key. The currently supported public key types are
+	// *rsa.PublicKey, *ecdsa.PublicKey and ed25519.PublicKey. (See x509 package)
+	SignCertificate(certificate *x509.Certificate, publicKey any) ([]byte, error)
```

Comment:
TIL that interface methods cannot have parameterized types. It would not compile. I actually found a way to remove the public key as a param altogether, so I will solve the conversation that way.